src/yaml-tools.js

```diff
@@ -110,12 +122,24 @@ yamlLibrary = {
 
   // doc: The SAM.YML parsed document
   // schema: the JSON schema to compare the doc to
-  checkDocAgainstSchema: function (doc, schema, failed, warning, info) {
-    this.doc = doc;
+  checkDocAgainstSchema: function (docIndex, schema, failed, warning, info) {
+    this.doc = this.docs[docIndex];
     var keys = [];
 
     this.recurseIntoArray(schema, keys, null, failed, warning, info);
+  },
+
+  loadDocData: function (docs) {
+    let docsArray = cleanDocsData(docs).split('---');
+
+    for (let i = 0; i < docsArray.length; i++) {
+      let yamlDoc = yaml.load(docsArray[i]);
+      // If a doc is empty depending on whether it's a single doc or included
+      // as part of several docs, the yaml parser will return a null or undefined value
+      // changing it to {} will force the parser to check against the schema doc
```

Comment:
```suggestion
      // If a doc is empty, whether it's a single doc or included as part of several docs
      // the yaml.load() will return a null or undefined value.  The parser, in a later call,
      // would return a successful validation for null or undefined items.  Changing it
      // to {} will force the parser to check against the schema doc and cause a failure.
```

Reply:
This doesn't seem clearer to me.

Reply:
It seems clearer to me as the person who didn't write it and knows the code in detail.